src/pages/AddPlane/AddPlane.js

```diff
@@ -0,0 +1,246 @@
+import { setAlert, setNotice } from '../../slices/appSlice/appSlice';
+
+// Functions
+// To be fully implemented later
+// import isValidImageUrl from '../../utils';
+
+// Actions
+import { createPlaneThunk } from '../../slices/planeSlice/planeAPI';
+
+const CreatePlaneSchema = Yup.object().shape({
+  name: Yup.string().min(2, 'Name is Too Short!').max(50, 'Name is Too Long!').required('Name is Required'),
+  plane_type: Yup.string().min(2, 'Type is Too Short!').max(50, 'Type is Too Long!').required('Plane type is Required'),
+  description: Yup.string().min(2, 'Name is Too Short!').required('Description is Required'),
+  price: Yup.number().positive().integer().required('Price is required'),
+  model: Yup.string().min(2, 'model is Too Short!').required('Model is Required'),
+  year_of_manufacture: Yup.date().required('Y-O-M is required'),
+  life_span: Yup.string().required('Lifes Span is required'),
+  fees: Yup.number().positive().integer().required('Fee is required'),
+});
+
+const AddPlane = () => {
+  const navigate = useNavigate();
+  const dispatch = useDispatch();
+  const token = useSelector((state) => state.auth.token);
+  const planeFormData = {
+    name: '',
+    plane_type: '',
+    description: '',
+    image: '',
+    price: 0,
+    model: '',
+    year_of_manufacture: '2000-04-06',
+    life_span: '',
+    fees: 0,
+  };
+  const [planeData, setPlaneData] = useState(planeFormData);
+  const [errors, setErrors] = useState({});
+  const [imageError, setImageError] = useState('');
+
+  const handleChange = (e) => {
+    setPlaneData({ ...planeData, [e.target.name]: e.target.value });
+  };
+
+  const handelSubmit = (e) => {
+    e.preventDefault();
+    setErrors({});
+    setImageError('');
+
+    const {
+      // eslint-disable-next-line camelcase
+      name, plane_type, description, price, model, year_of_manufacture, life_span, fees,
+    } = planeData;
+
+    // To be fully implemented later
+
+    // if (isValidImageUrl(planeData.image) !== true) {
+    //   setImageError('Invalid image url');
+    //   return;
+    // }
+    CreatePlaneSchema.validate({
+      // eslint-disable-next-line camelcase
+      name, plane_type, description, price, model, year_of_manufacture, life_span, fees,
+    }, { abortEarly: false })
+      .then(() => {
+        const data = {
+          planeData,
+          token,
+        };
+
+        dispatch(createPlaneThunk(data)).then((res) => {
+          if (res.error) {
+            dispatch(setAlert(res.payload));
+          } else {
+            dispatch(setNotice('New Plane Created Successfully'));
+            navigate('/');
+          }
+        });
+      }).catch((err) => {
+        const newErrors = {};
+        err.inner.forEach((error) => {
+          newErrors[error.path] = error.message;
+        });
+        setErrors(newErrors);
+      });
+  };
+
+  return (
+    <section className="bg-gray-50 dark:bg-gray-900 ">
```

Comment:
- [ ] Kindly fix the stying here because it s broken on my browser:


https://user-images.githubusercontent.com/95412943/230714268-48993cbb-bb91-4d4d-8b18-c055c9a9cc14.mp4